scaletest/templates/kubernetes-minimal/main.tf

```diff
@@ -0,0 +1,164 @@
+  url          = "http://localhost:1235"
+  subdomain    = true
+  share        = "authenticated"
+}
+
+resource "coder_app" "ws_discard" {
+  agent_id     = coder_agent.m.id
+  slug         = "wsdi" # Short slug so URL doesn't exceed limit: https://wsdi--main--scaletest-UN9UmkDA-0--scaletest-SMXCCYVP-0--apps.big.cdr.dev
+  display_name = "WebSocket Discard"
+  url          = "http://localhost:1236"
+  subdomain    = true
+  share        = "authenticated"
+}
+
+resource "kubernetes_deployment" "main" {
+  count = data.coder_workspace.me.start_count
+  metadata {
+    name      = "coder-${lower(data.coder_workspace.me.owner)}-${lower(data.coder_workspace.me.name)}"
+    namespace = "coder-big"
+    labels = {
+      "app.kubernetes.io/name"     = "coder-workspace"
+      "app.kubernetes.io/instance" = "coder-workspace-${lower(data.coder_workspace.me.owner)}-${lower(data.coder_workspace.me.name)}"
+      "app.kubernetes.io/part-of"  = "coder"
+      "com.coder.resource"         = "true"
+      "com.coder.workspace.id"     = data.coder_workspace.me.id
+      "com.coder.workspace.name"   = data.coder_workspace.me.name
+      "com.coder.user.id"          = data.coder_workspace.me.owner_id
+      "com.coder.user.username"    = data.coder_workspace.me.owner
+    }
+  }
+  spec {
+    replicas = 1
+    selector {
+      match_labels = {
+        "app.kubernetes.io/instance" = "coder-workspace-${lower(data.coder_workspace.me.owner)}-${lower(data.coder_workspace.me.name)}"
+      }
+    }
+    strategy {
+      type = "Recreate"
+    }
+    template {
+      metadata {
+        labels = {
+          "app.kubernetes.io/name"     = "coder-workspace"
+          "app.kubernetes.io/instance" = "coder-workspace-${lower(data.coder_workspace.me.owner)}-${lower(data.coder_workspace.me.name)}"
+        }
+      }
+      spec {
+        security_context {
+          run_as_user = "1000"
+          fs_group    = "1000"
+        }
+        container {
+          name              = "dev"
+          image             = "docker.io/codercom/enterprise-minimal:ubuntu"
+          image_pull_policy = "IfNotPresent"
+          command           = ["sh", "-c", coder_agent.m.init_script]
+          security_context {
+            run_as_user = "1000"
+          }
+          env {
+            name  = "CODER_AGENT_TOKEN"
+            value = coder_agent.m.token
+          }
+          resources {
+            requests = {
+              "cpu"    = "100m"
+              "memory" = "320Mi"
+            }
+            limits = {
+              "cpu"    = "100m"
+              "memory" = "320Mi"
+            }
+          }
+        }
+
+        affinity {
+          node_affinity {
+            required_during_scheduling_ignored_during_execution {
+              node_selector_term {
+                match_expressions {
+                  key      = "cloud.google.com/gke-nodepool"
+                  operator = "In"
+                  values   = ["big-workspaces", "big-workspaces2"]
```

Comment:
We might want to make this a parameter instead so it's easier to control the node affinity key/value.

Reply:
I verified this, and it looks like we need to add the parameter to `scaletest-runner`, and then modify the `scaletest create-workspaces` to accept the parameter. It sounds to me more like a follow-up?

Reply:
TBH it could just be hard-coded in the template but folks who just use the template as-is are likely going to run into issues. Maybe add a note about having to edit the node affinity to match your environment.

Reply:
Alright, I added a template variable and dropped a note in docs.